backend/src/components/permissionHelpers.js

```diff
@@ -0,0 +1,78 @@
+const log = require('npmlog');
+const utils = require('./utils');
+
+
+const permissionHelpers = {
+  // TODO: this is likely soon to be refactored out, as we will be pulling acronyms from the DB, not from access roles
+  // Returns only app acronym based roles
+  filterAppAcronymRoles: roles => roles.filter(role => !role.match(/offline_access|uma_authorization|WEBADE_CFG_READ|WEBADE_CFG_READ_ALL|WEBADE_PERMISSION|WEBADE_PERMISSION_NROS_DMS/)),
+
+  /**
+  * Is this call allowed to be made for the acronym it's being made for?
+  * @param {string} token - The user's token from the request.
+  * @param {string} acronym - The app acronym.
+  * @returns {string} Error Message - Undefined if permitted, a string of the error message to return if not permitted.
+  */
+  checkAcronymPermission: (token, applicationAcronym) => {
+    // TODO: a lot of this role checking is duplicate, but as we will be moving acronym management to the DB soon all this will need to be refactored anyways
+
+    // Get Acronyms from user token
+    let acronyms = [];
+    const roles = token.realm_access.roles;
+    if (typeof roles === 'object' && roles instanceof Array) {
+      acronyms = permissionHelpers.filterAppAcronymRoles(roles);
+    }
+
+    // Do they have access to the Acronym they are trying to POST
+    const appAcronym = applicationAcronym;
+    if (!acronyms.includes(applicationAcronym)) {
+      log.verbose(`User not authorized for acronym ${applicationAcronym}. Token: ${utils.prettyStringify(token)}`);
+      return `User lacks permission for '${appAcronym}' acronym`;
+    }
+  },
+
+  /**
+  * Is this call allowed to post the WebADE config details?
+  * @param {string} token - The user's token from the request.
+  * @param {string} configForm - The form posted to the post enpoint.
+  * @param {string} acronymData - The acronym details from the getok db.
+  * @param {string} desiredUserRoles - Specific user permissions that are needed for the operation to check.
+  * @returns {string} Error Message - Undefined if permitted, a string of the error message to return if not permitted.
+  */
+  checkWebAdePostPermissions: (token, configForm, acronymData, desiredUserRoles = []) => {
+    try {
+      // Do they have access to the Acronym they are trying to POST
+      let acronymAccessError = permissionHelpers.checkAcronymPermission(token, configForm.applicationAcronym);
+      if (acronymAccessError) {
+        return acronymAccessError;
+      }
+
+      // Do they have the specified scopes
+      let roleError = undefined;
+      desiredUserRoles.forEach((role) => {
+        if (!token.realm_access.roles.includes(role)) {
+          roleError = `User is not permitted to submit WebADE config, missing role ${role}`;
+          return;
+        }
+      });
+      if (roleError) {
+        return roleError;
+      }
```

Comment:
Yeah this is intended. It's not a list of missing roles, it's just a guard against at least one role being missing. As soon as it finds one lacking it can return from the iteration and fail the permission check. Not a list, just a boolean condition essentially.

Reply:
If we want just a simple boolean block, the following would be able to achieve it:
```
if(!desiredUserRoles.every(role => token.realm_access.roles.includes(role))) {
  return `User is not permitted to submit WebADE config, missing some roles`
}
```
If we need to return in the message which role is missing the logic would need to be slightly more complex.